x/gamm/pool-models/balancer/pool.go

```diff
@@ -275,6 +275,16 @@ func (p Pool) parsePoolAssets(tokensA sdk.Coins, tokenBDenom string) (
 	return tokensA[0], Aasset, Basset, nil
 }
 
+func (p Pool) parsePoolAssetsForSwap(tokenA sdk.Coin, tokenBDenom string) (
+	Aasset PoolAsset, Basset PoolAsset, err error,
+) {
+	Aasset, Basset, err = p.parsePoolAssetsByDenoms(tokenA.Denom, tokenBDenom)
+	if err != nil {
+		return PoolAsset{}, PoolAsset{}, err
+	}
+	return Aasset, Basset, nil
+}
```

Comment:
This looks like it just wraps parsePoolAssetsByDenoms right? Why not just call parsePoolAssetsByDenoms directly?